File: src/main/java/tomeko/hymod/config/ModMenuIntegration.java
package tomeko.hymod.config;

//? if >= 1.21.9 {
/*import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.gui.screens.Screen;
import org.polyfrost.oneconfig.utils.v1.dsl.ScreensKt;

@Environment(EnvType.CLIENT)
public class ModMenuIntegration implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return (Screen parent) -> ScreensKt.createScreen(HyModConfig.INSTANCE);
    }
}
*///?}

File: src/main/kotlin/tomeko/hymod/config/ModMenuIntegration.kt
package tomeko.hymod.config

//? if >= 1.21.9 {
/*import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import net.minecraft.client.gui.screens.Screen
import org.polyfrost.oneconfig.utils.v1.dsl.createScreen

class ModMenuIntegration : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return { _: Screen -> HyModConfig.createScreen() }
    }
}
*///?}